app/controllers/cart_products_controller.rb

```diff
@@ -0,0 +1,21 @@
+class CartProductsController < ApplicationController
+  def create
+    products_in_cart = current_user.cart.products
+
+    if products_in_cart.map(&:id).include?(params[:product_id].to_i)
```

Comment:
Here, we check to see if the products that currently exist in the cart already include the item we're trying to add to our cart.